airbyte-ci/connectors/pipelines/pipelines/contexts.py

```diff
@@ -476,6 +488,190 @@ async def __aexit__(
+        return self._secrets_dir
+
+    @secrets_dir.setter
+    def secrets_dir(self, secrets_dir: Directory):  # noqa D102
+        self._secrets_dir = secrets_dir
+
+    @property
+    def updated_secrets_dir(self) -> Directory:  # noqa D102
+        return self._updated_secrets_dir
+
+    @updated_secrets_dir.setter
+    def updated_secrets_dir(self, updated_secrets_dir: Directory):  # noqa D102
+        self._updated_secrets_dir = updated_secrets_dir
+
+    @property
+    def should_save_updated_secrets(self) -> bool:  # noqa D102
+        return self.use_remote_secrets and self.updated_secrets_dir is not None
+
+
+    async def __aexit__(
+        self, exception_type: Optional[type[BaseException]], exception_value: Optional[BaseException], traceback: Optional[TracebackType]
+    ) -> bool:
+        """Perform teardown operation for the CDKContext.
+
+        On the context exit the following operations will happen:
+            - Write a test report in JSON format locally and to S3 if running in a CI environment
+            - Update the commit status check on GitHub if running in a CI environment.
+        It should gracefully handle the execution error that happens and always upload a test report and update commit status check.
+        Args:
+            exception_type (Optional[type[BaseException]]): The exception type if an exception was raised in the context execution, None otherwise.
+            exception_value (Optional[BaseException]): The exception value if an exception was raised in the context execution, None otherwise.
+            traceback (Optional[TracebackType]): The traceback if an exception was raised in the context execution, None otherwise.
+        Returns:
+            bool: Whether the teardown operation ran successfully.
+        """
+        self.stopped_at = datetime.utcnow()
+        self.state = self.determine_final_state(self.report, exception_value)
+        if exception_value:
+            self.logger.error("An error got handled by the ConnectorContext", exc_info=True)
+        if self.report is None:
+            self.logger.error("No test report was provided. This is probably due to an upstream error")
+            self.report = ConnectorReport(self, [])
+
+        if self.should_save_updated_secrets:
+            await secrets.upload(self)
+
+        self.report.print()
+
+        if self.should_save_report:
+            await self.report.save()
+
+        if self.report.should_be_commented_on_pr:
+            self.report.post_comment_on_pr()
+
+        await asyncify(update_commit_status_check)(**self.github_commit_status)
+
+        if self.should_send_slack_message:
+            await asyncify(send_message_to_webhook)(self.create_slack_message(), self.reporting_slack_channel, self.slack_webhook)
+
+        # Supress the exception if any
+        return True
+
+    def create_slack_message(self) -> str:
+        raise NotImplementedError
+    
+
+class PublishCDKContext(CDKContext):
```

Comment:
Do you really need to add it in this PR as the publish flow is not implemented?

Reply:
Probably not, if we decide to move forward with this approach I'll probably remove it to descope the pr